src/Logger.cpp

```diff
@@ -30,51 +62,57 @@ void cLogger::InitiateMultithreading()
 
 
 
-void cLogger::LogSimple(const AString & a_Message, eLogLevel a_LogLevel)
+void cLogger::LogSimple(std::string_view a_Message, eLogLevel a_LogLevel)
 {
-	time_t rawtime;
-	time(&rawtime);
+	fmt::memory_buffer Buffer;
+	WriteLogOpener(Buffer);
+	fmt::format_to(Buffer, "{0}\n", a_Message);
+	LogLine(std::string_view(Buffer.data(), Buffer.size()), a_LogLevel);
+}
+
 
-	struct tm * timeinfo;
-	#ifdef _MSC_VER
-		struct tm timeinforeal;
-		timeinfo = &timeinforeal;
-		localtime_s(timeinfo, &rawtime);
-	#else
-		timeinfo = localtime(&rawtime);
-	#endif
-
-	AString Line;
-	#ifdef _DEBUG
-		Printf(Line, "[%04llx|%02d:%02d:%02d] %s\n", static_cast<UInt64>(std::hash<std::thread::id>()(std::this_thread::get_id())), timeinfo->tm_hour, timeinfo->tm_min, timeinfo->tm_sec, a_Message);
-	#else
-		Printf(Line, "[%02d:%02d:%02d] %s\n", timeinfo->tm_hour, timeinfo->tm_min, timeinfo->tm_sec, a_Message);
-	#endif
 
 
+
+void cLogger::LogLine(std::string_view a_Line, eLogLevel a_LogLevel)
+{
 	cCSLock Lock(m_CriticalSection);
 	for (size_t i = 0; i < m_LogListeners.size(); i++)
 	{
-		m_LogListeners[i]->Log(Line, a_LogLevel);
+		m_LogListeners[i]->Log(a_Line, a_LogLevel);
 	}
 }
 
 
 
 
 
-void cLogger::vLogPrintf(const char * a_Format, eLogLevel a_LogLevel, fmt::printf_args a_ArgList)
+void cLogger::LogPrintf(
+	std::string_view a_Format, eLogLevel a_LogLevel, fmt::printf_args a_ArgList
+)
 {
-	LogSimple(vPrintf(a_Format, a_ArgList), a_LogLevel);
+	fmt::memory_buffer Buffer;
+	WriteLogOpener(Buffer);
+	fmt::printf(Buffer, fmt::to_string_view(a_Format), a_ArgList);
+	fmt::format_to(Buffer, "\n");
+
+	LogLine(std::string_view(Buffer.data(), Buffer.size()), a_LogLevel);
```

Comment:
This is the most important change. Writing everything by appending into `Buffer`, then using a `std::string_view` to avoid copies when calling the log listeners.